src/counter.cpp

```diff
@@ -0,0 +1,59 @@
+#include "counter.h"
+#include <cstring>
+extern "C" {
+    #include <tree_sitter/api.h>
+}
+
+Counter::Counter(Parser* parser, Countable countable, std::string feature, std::vector<std::string>& files)
+  : parser(parser)
+  , countable(countable)
+  , feature(feature)
+  , files(files)
+{}
+
+void Counter::count_feature() {
+    for (size_t i = 0; i < files.size(); i++)
+    {
+      TSTree *tree = parser->parse_file(files[i]);
+      TSNode root_node = ts_tree_root_node(tree);
+      TSTreeCursor cursor = ts_tree_cursor_new(root_node);
+      while (true)
+      {
+          if (ts_tree_cursor_goto_first_child(&cursor) || ts_tree_cursor_goto_next_sibling(&cursor)) {
+            TSNode cur = ts_tree_cursor_current_node(&cursor);
+            const char* type = ts_node_type(cur);
+            if (!strcmp("identifier", type) && is_parent_call) {
```

Comment:
It's generally safer to use `strncmp()` instead of `strcmp()`.  The same comment applies elsewhere throughout this file.

Reply:
If [this](https://stackoverflow.com/a/30190652/12458945) is the reason that you suggested the change would it be sufficient to change the `strcmp()` to `strncmp()` in places where we are using user input as an parameter to `strcmp` function.